R/shinyGadget.R

```diff
@@ -1,29 +1,98 @@
-basicProjectGadget <- function() {
-
-ui <- miniUI::miniPage(
-  miniUI::gadgetTitleBar("Create Basic pRoject"),
-  miniUI::miniContentPanel(
-    shiny::inputPanel(
-      shiny::textInput("name", "Project Name")
-    ),
-    shiny::inputPanel(
-      shiny::checkboxInput("readme", "Create README", value = TRUE),
-      shiny::checkboxInput("git", "Use Git", value = TRUE),
-      shiny::checkboxInput("travis", "Use Travis", value = TRUE),
-      shiny::checkboxInput("packrat", "Use Packrat", value = TRUE)
+projectGadget <- function() {
+
+  ui <- miniUI::miniPage(
+    miniUI::gadgetTitleBar("Create pRoject"),
+    miniUI::miniContentPanel(
+      shiny::inputPanel(
+        shiny::selectInput(
+          "projectType", label = "Project Type",
+          choices = c("Basic","Analysis","Package","Training"),
+          selected = "Basic"
+        ),
+        shiny::textInput("name", "Project Name")
+      ),
+      shiny::uiOutput("ui"),
+      shiny::conditionalPanel(
+        condition = "input.projectType != 'Package'",
+        shiny::inputPanel(
+          shiny::h4("Best Practices"),
+          shiny::br(),
+          shiny::checkboxInput("readme", "Create README", value = TRUE),
+          shiny::checkboxInput("git", "Use Git", value = TRUE),
+          shiny::checkboxInput("travis", "Use Travis", value = TRUE),
+          shiny::checkboxInput("packrat", "Use Packrat", value = TRUE)
+        )
+      )
     )
   )
-)
 
-server <- function(input, output, session) {
+  server <- function(input, output, session) {
 
-  shiny::observeEvent(input$done, {
-    createBasicProject(name = input$name, readme = input$readme,
-                       git = input$git, travis = input$travis,
-                       packrat = input$packrat)
-    shiny::stopApp()
-  })
-}
+    output$ui <- renderUI({
+
+      switch(input$projectType,
+
+        "Basic"     = shiny::div(),
+        "Analysis"  = shiny::inputPanel(
+                        shiny::selectInput("dirs", label = "Directories",
```

Comment:
I had the same thought but was lazy so just did it the "neat" way to start with.

In general (not just in this instance) I think it's probably a good idea to try and make things as flexible as possible. But one also has to balance that with efficiency - e.g. we could allow custom text inputs here, but I'd argue that typing directory names into the gadget isn't really much more efficient than typing them out in the function call directly or just creating them manually. So on a case by case basis, that flexibility isn't really worth much.

If more and/or different directory names are needed on a repeated basis then I'd say this is another potentially good candidate for the yaml template we discussed previously. Then this directory list could be populated from there which should give the best of both worlds (flexibility for repeated cases + better efficiency)